website/content/gateway/api_reference/resources/debit_refunds/debit_refunds.md

```diff
@@ -0,0 +1,19 @@
+---
+title: "debit_refunds"
+date: 2023-01-05T13:37:00+01:00
+anchor: "debit_refunds"
+weight: 178
+---
+#### Debit refunds
+To refund money to a cardholder's bank account that has been transferred by a debit you make a new refund resource. Only a single refund can be made for a debit and it can only be for the full amount.
+
+```shell
+POST https://gateway.clearhaus.com/debits/:id/refunds
+```
+
+##### Parameters
+No parameters are needed to make a new refund transaction.
+
+{{% notice %}}
+**Notice**: A refund of a Visa AFT debit might be declined by Visa or the issuer if the debit age is 24 hours or more, or the debit has cleared. The transaction gateway does not track the clearing of the debit, i.e. it relies on Visa and the issuer to decline late refunds.
```

Comment:
It's not clear to me what the consequence of 

> The transaction gateway does not track the clearing of the debit

is.

Would this mean that an approved transaction isn't necessarily successfully cleared, and it would be on me to later reconcile with the merchant API whether the funds are deducted from my account or not?

Reply:
When an issuer, or Visa on behalf of an issuer via STIP (stand in), approves a debit (AFT), we expect the debit to be cleared eventually. Because this is SMS, not DMS, we don't send a separate clearing "request" upstream for a debit.
As with DMS, it would probably be incorrect/naive to expect that nothing can go wrong with the clearing, but we strongly expect it to go through.
When a debit is eventually cleared, the merchant API will know about it. The transaction gateway will not.
Visa states that a debit refund (AFTR) must be used to refund within the first 24 hours of the debit but before the clearing of the transaction. After this vague limit another mechanism ought to be used (we don't plan to support that at this time).
Since the transaction gateway doesn't know about clearing, at least currently, we have decided for now to let upstream (Visa/issuer) decide whether a "late" debit refund is approved or not.
I hope this makes matters a bit more clear.

Reply:
Could this be an improvement?

```suggestion
**Notice**: In almost all cases, an approved debit is eventually cleared by Visa and the issuer. It should be noted that Visa or the issuer might decline a debit refund with the reasoning that 24 hours have passed since the debit or that the debit has been cleared. There is no dedicated signaling for this, so the transaction has no dedicated status code. We recommend that a debit refund is created as soon as possible, in case a debit needs to be refunded.
```

cc: @ct-clearhaus @clearhaus/transaction-compliance

Reply:
> with DMS, it would probably be incorrect/naive to expect that nothing can go wrong with the clearing, but we strongly expect it to go through

Maybe I'm just in error, but I, as an integrator, would assume that liability to be on Clearhaus, once the gateway responds "OK" to the capture. Basically that's the guarantee extended from the authorization: the money is safe within  the ruleset of the scheme and your acquirer agreement. That's why merchants dare ship the goods on capture or otherwise debit a customer.
If the clearing fails, I'd expect that to be opaque to the merchant. E.g. not something we'd need to backfill from the merchant API and notify the merchant somehow that they won't get their funds anyway.

So the question is, I guess, is that generally wrong, or if not, is it likewise with AFTs, or would merchants assume some new type of risk that we'd need to have a technical answer to?

On another, more curiosity note, do you have a guess why "might be declined by Visa or the issuer if ... the debit has cleared" is the case? That makes their use quite limited, right? And very different from "capture refunds"?

Reply:
@tjconcept You bring up an interesting issue and some good questions. Sincere thank you for doing so.

I am not an expert on these matters, so I have discussed it with a few colleagues.

Rejects connected to clearing can also happen for DMS, i.e. for our current captures and refunds, and we do not reflect that in the transaction gateway API either. In other words, debits/AFTs do not pose a new challenge here. Handling these rejects is a partly manual process, where the involved parties' state is reverted. Fortunately, we do not see many rejections.

> would merchants assume some new type of risk that we'd need to have a technical answer to?

With regard to risk, we do not assume debits/AFTs to be different from the current authorizations, captures and refunds (DMS). However, since we have no history and experience some difficulty in getting firm statements from Visa, we cannot confirm it.

> do you have a guess why "might be declined by Visa or the issuer if ... the debit has cleared" is the case? That makes their use quite limited, right? And very different from "capture refunds"?

We are not entirely sure, however, as for AFT use cases it seems natural that you deliver quickly and, if that fails, you also refund quickly. We imagine within seconds or at most a few minutes.

An option to refund later exists in Visa's machinery, however, this option is another transaction type that is apparently badly supported among issuers as many or all them process the transactions manually with quite a delay (probably several days or even weeks). We do not support this option.

Reply:
Okay, thanks. I'll bring it up with our partner contact when discussing our beta merchant's use case. In short, we either need to invest significantly in as-recent-as-possible reconciliation to build our own "notification" of clearing issues (as we could be losing uncapped volumes) while we test and get a sense of the risk, or Clearhaus needs to cover it one way or another.
I'd argue any day for moving that risk upstream as far as possible (making all downstreams simpler), but that's unfortunately "business" and not just rational, I suppose.

> is another transaction type

Oh well 😅